src/Core/Notifications/IDisabledNotificationsStorage.cs

```diff
@@ -32,15 +40,95 @@ public interface IDisabledNotificationsStorage
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class DisabledNotificationsStorage : IDisabledNotificationsStorage
     {
-        public void DisableNotification(string id)
+        private readonly IVsVersionProvider vsVersionProvider;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+
+        private string filePath => GetFilePath();
+
+        private DisabledNotifications disabledNotifications = null;
+        private readonly object lockObject = new object();
+
+        [ImportingConstructor]
+        public DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger) : this(vsVersionProvider, logger, new FileSystem())
         {
-            // todo
+
         }
 
+        internal /*for testing*/ DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger, IFileSystem fileSystem)
+        {
+            this.vsVersionProvider = vsVersionProvider;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public void DisableNotification(string id)
+        {
+            lock (lockObject)
+            {
+                if (IsNotificationDisabled(id))
+                {
+                    return;
+                }
+                if (disabledNotifications == null)
+                {
+                    return;
+                }
+
+                disabledNotifications.AddNotification(id);
+                SaveNotifications();
+            }
+        }
         public bool IsNotificationDisabled(string id)
         {
-            // todo
-            return false;
+            lock (lockObject)
+            {
+                if (disabledNotifications == null)
+                {
+                    disabledNotifications = ReadDisabledNotifications();
+                }
+                if (disabledNotifications == null)
+                {
+                    logger.WriteLine("Couldn't find disabled notification config. Notification will be showed");
+                    return false;
+                }
+                return disabledNotifications.Notifications.Any(n => n.Id == id);
+            }
+        }
+
+        private DisabledNotifications ReadDisabledNotifications()
+        {
+            try
+            {
+                var fileContent = fileSystem.File.ReadAllText(filePath);
```

Comment:
This will throw if the file doesn't exist, which is normal and expected situation if the user has never disabled a notification.

It would be better to handle the file-not-found case separately e.g. something like:

```
if (!File.Exists(filePath))
{
  logger.LogDebug($"[Notifications] Disabled notifications file does not exist. File: {filePath}");
  return new DisabledNotifications();
}
```